main/src/main/scala/sbt/Defaults.scala

```diff
@@ -351,12 +369,14 @@ object Defaults extends BuildCommon {
           crossPaths.value
         )
     },
-    unmanagedSources := collectFiles(
-      unmanagedSourceDirectories,
-      includeFilter in unmanagedSources,
-      excludeFilter in unmanagedSources
-    ).value,
-    watchSources in ConfigGlobal ++= {
+    unmanagedSources := FileManagement
+      .collectFiles(
+        unmanagedSourceDirectories,
+        includeFilter in unmanagedSources,
+        excludeFilter in unmanagedSources
+      )
+      .value,
+    watchSources in ConfigGlobal := (watchSources in ConfigGlobal).value ++ {
```

Comment:
> Using ++= prevents these from being overridden by the user without doing
some nasty runtime manipulations.

I'm confused by this?  How is this line different than the original?

Reply:
Welcome back, @dwijnand :). I added this line because it was impossible for me to remove the default watchSources in the CloseWatchPlugin because of the use of `++=`. In a plugin or a build, if you do something like `watchSources in ConfigGlobal := Nil`, you will still end up with the default watch sources. I had to manually remove these in the CloseWatchPlugin in a very obscure way: https://github.com/swoval/swoval/blob/0d4882e87e0d1deb710338e3d55bed17d772bbe4/plugin/src/main/scala/com/swoval/watchservice/CloseWatchPlugin.scala#L278

Reply:
OK, the reason for that is probably because in `Defaults` it's defined in `sourceConfigPaths` (i.e applied per configuration), while in `CloseWatchPlugin` you were probably trying to define it in `globalSettings`.  So it's an ordering thing of which definition in Global scope wins.

But that doesn't 100% explain why `k ++= x` transformed to `k := k.value ++ x` aren't the same.  Unless, of course, they're subtly different when sbt does it's setting re-ordering thing as it processes them (I forget where).

🤷‍♂️

Reply:
I think your intuition is right.  I was just grasping at straws trying to come up with plausible reasons why the change made a difference.  But if you're saying you didn't test it, then I'm more confident that it changed nothing. 😄

`:=` is actually interesting because it'll even suppress setting mistakes, for instance:

```scala
val someSources = settingKey[Seq[String]]("")

someSources ++= Seq("a", "b", "c") // or 'someSources := someSources.value ++ ..'
someSources := Seq("d")
```

results in `Seq("d")`, but without the `:=` line it results in

```scala
[error] Reference to undefined setting:
[error]
[error]   someSources from someSources (/s/t-test-order/build.sbt:4)
```

Reply:
Oh and large parts of my understand about scopes and settings comes from experimentation 🙂